h5io/_h5io.py

```diff
@@ -68,15 +69,21 @@ def write_hdf5(fname, data, overwrite=False, compression=4,
         this your package name, e.g. ``'mnepython'``.
     """
     h5py = _check_h5py()
-    if op.isfile(fname) and not overwrite:
-        raise IOError('file "%s" exists, use overwrite=True to overwrite'
-                      % fname)
+    mode = 'w'
+    if op.isfile(fname):
+        if overwrite is False:
+            raise IOError('file "%s" exists, use overwrite=True to overwrite'
+                          % fname)
+        if overwrite is 'update':
```

Comment:
don't use `is` for str comparison, use `if overwriite == 'update'`. Also add check for overwrite being `update` if of type `string_types`. So something like this:
```
if op.isfile(fname):
    if isinstance(overwrite, string_types):
        if overwrite != 'update':
            raise ValueError('overwrite must be "update" or a bool')
        mode = 'a'
    elif not overwrite:
        raise IOError(...)
```